sessionctx/variable/tidb_vars.go

```diff
@@ -751,6 +751,7 @@ var (
 		ReportIntervalSeconds: atomic.NewInt64(DefTiDBTopSQLReportIntervalSeconds),
 	}
 	EnableLocalTxn = atomic.NewBool(DefTiDBEnableLocalTxn)
+	LockUniqueKeys = atomic.NewBool(true)
```

Comment:
@jackysp @coocood @youjiali1995 @lysu 
Or do we need to change it to `false` to change the behaviour back with v4.0.9 and earlier versions, but this could affect the users using v5.0 release versions if they rely on the new behaviours.